nemo/collections/asr/models/confidence_ensemble.py

```diff
@@ -12,25 +12,139 @@
+class ConfidenceSpec:
+    exclude_blank: bool
+    aggregation: str
+    confidence_type: str
+    alpha: float
+
+    def to_confidence_config(self) -> ConfidenceConfig:
+        """Converts confidence spec to the confidence config.
+
+        Internally, the tuning procedure uses this "spec" objects as they
+        are more aligned with how things are implemented. But when it's time
+        to save the models or call transcribe, we need to use the proper
+        object of type ``ConfidenceConfig``.
+        """
+        if self.confidence_type == 'max_prob':
+            name = 'max_prob'
+            entropy_type = 'tsallis'  # can be any
+            entropy_norm = 'lin'  # can be any
+        else:
+            name, entropy_type, entropy_norm = self.confidence_type.split("_")
+        return ConfidenceConfig(
+            exclude_blank=self.exclude_blank,
+            aggregation=self.aggregation,
+            method_cfg=ConfidenceMethodConfig(
+                name=name, entropy_type=entropy_type, temperature=self.alpha, entropy_norm=entropy_norm,
+            ),
+        )
+
+
+def get_filtered_logprobs(hypothesis: Hypothesis, exclude_blank: bool) -> torch.Tensor:
+    """Returns logprobs from the hypothesis object with optional blanks filter.
+
+    This function supports both CTC and Transducer hypotheses. Will place the
+    logprobs on GPU if it's available.
+
+    Args:
+        hypothesis: generated hypothesis as returned from the transcribe
+            method of the ASR model.
+        exclude_blank: whether to filter out all ``<blank>`` tokens.
+
+    Returns:
+        torch.Tensor: of shape [S, V], where S is (filtered) sequence length and
+        V is the vocabulary size.
+    """
+    if isinstance(hypothesis.alignments, list):  # Transducer
+        filtered_logprobs = []
+        for alignment in hypothesis.alignments:
+            for align_elem in alignment:
+                if exclude_blank and align_elem[1].item() != align_elem[0].shape[-1] - 1:
+                    filtered_logprobs.append(align_elem[0])
+                filtered_logprobs.append(align_elem[0])
+        if not filtered_logprobs:  # for the edge-case of all blanks
+            filtered_logprobs.append(align_elem[0])
+        filtered_logprobs = torch.stack(filtered_logprobs)
+        if torch.cuda.is_available():  # by default logprobs are placed on cpu in nemo
+            filtered_logprobs = filtered_logprobs.cuda()
+    else:  # CTC
+        logprobs = hypothesis.y_sequence
+        if torch.cuda.is_available():  # by default logprobs are placed on cpu in nemo
+            logprobs = logprobs.cuda()
+        if exclude_blank:  # filtering blanks
+            labels = logprobs.argmax(dim=-1)
+            filtered_logprobs = logprobs[labels != logprobs.shape[1] - 1]
+        else:
+            filtered_logprobs = logprobs
+    return filtered_logprobs
+
+
+def compute_confidence(hypothesis: Hypothesis, confidence_cfg: ConfidenceConfig) -> float:
```

Comment:
Wouldn't it be better for these functions to all go into the confidence mixin then ? And here you just import / instantiate ? Rather than have two separate implementation